### PR TITLE
Fix 8.0.13 binlog encryption

### PR DESCRIPTION
**Pull request by robgolebiowski**
m_fdle removed from rpl_binlog_sender.cc/.h . It was used in binlog
encryption for remembering crypto information for binlog sender thread.
Now as crypto information is a part of binlog reader it is no longer
needed.

Crypto related function was removed from Format_descriptor_log_event and
moved to binlog reader. Now Encryption is an attribiute of
Binlog_event_data_istream.

Removed implementation of Start_encryption_log_event::do_apply_event as
start_encryption is now only done in binlog reader.

reset_crypto is called when new binlog file is opened.

### CODE REVIEW

File: mysql-test/suite/rpl_encryption/r/encrypted_master_unload_keyring.result [hunk 1]
```diff
@@ -28,7 +28,7 @@ Pattern "table1_encryption" not found
 # Part 2: restart master, without binlog encryption and without keyring plugin
 #################################################################################
 call mtr.add_suppression("Error reading GTIDs from binary log");
-call mtr.add_suppression("Error initializing decryption while reading GTIDs from binary log");
+call mtr.add_suppression("Failed to initlalize binlog encryption. Please make sure correct keyring is loaded.");
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rpl_encryption/r/encrypted_master_unload_keyring.result]: "initialize"

File: sql/binlog_istream.cc [hunk 2]
```diff
@@ -50,6 +50,11 @@ const char *Binlog_read_error::get_str() const {
     case BAD_BINLOG_MAGIC:
       return "Binlog has bad magic number;  It's not a binary log file "
              "that can be used by this version of MySQL";
+    case DECRYPT_INIT_FAILURE:
+      return "Failed to initlalize binlog encryption. Please make sure correct keyring "
```
Comment (laurynas-biveinis) [inline sql/binlog_istream.cc]: "initialize"

Comment (laurynas-biveinis) [inline sql/binlog_istream.cc]: "Please make sure that the correct keyring plugin is loaded"

**Review by laurynas-biveinis — APPROVED** [review]
Please reference PS-5105 in the commit message header
OK to push after minor comment addressing, no full test round necessary

**Comment by robgolebiowski** [comment]
JR => https://ps.cd.percona.com/view/8.0/job/percona-server-8.0-param/106/
JR => https://ps2.cd.percona.com/view/8.0/job/percona-server-8.0-param/6/

**Pull request merged**